elements/pl-string-input/pl-string-input.mustache

```diff
@@ -0,0 +1,74 @@
+        <span class="input-group-append">
+            {{#suffix}}<span class="input-group-text">{{suffix}}</span>{{/suffix}}
+            <a class="btn btn-light border" role="button" data-toggle="popover" data-html="true" title="Number" data-content="{{info}}" data-placement="auto" data-trigger="focus" tabindex="0">
+                <i class="fa fa-question-circle" aria-hidden="true"></i>
+                {{#correct}}&nbsp;<span class="badge badge-success"><i class="fa fa-check" aria-hidden="true"></i> 100%</span>{{/correct}}
+                {{#partial}}&nbsp;<span class="badge badge-warning"><i class="fa fa-circle-o" aria-hidden="true"></i> {{partial}}%</span>{{/partial}}
+                {{#incorrect}}&nbsp;<span class="badge badge-danger"><i class="fa fa-times" aria-hidden="true"></i> 0%</span>{{/incorrect}}
+            </a>
+        </span>
+    </span>
+{{#inline}}</span>{{/inline}}
+{{/question}}
+
+
+{{#submission}}
+{{#inline}}<span class="d-inline-block">{{/inline}}
+{{#parse_error}}
+<script>
+    $(function(){
+        $('#pl-string-input-submission-{{uuid}} [data-toggle="popover"]').popover({
+            container: 'body',
+            template: '<div class="popover pl-string-input-popover" role="tooltip"><div class="arrow"></div><h3 class="popover-header"></h3><div class="popover-body"></div></div>',
+        });
+    });
+</script>
+<span id="pl-string-input-submission-{{uuid}}">
+    {{#label}}<span>{{label}}</span>{{/label}}
+    <a class="btn btn-secondary btn-sm" role="button" data-placement="auto" data-trigger="focus" data-toggle="popover" data-html="true" title="Error" tabindex="0" data-content="{{parse_error}}{{#raw_submitted_answer}}<hr><h5><small class='text-uppercase'>What you submitted</small></h5><pre>{{raw_submitted_answer}}</pre>{{/raw_submitted_answer}}">INVALID <span><i class="fa fa-question-circle" aria-hidden="true"></i></span></a>
+</span>
+{{/parse_error}}
+{{^parse_error}}
+{{#label}}<span>{{label}}</span>{{/label}}
+<samp>{{a_sub}}</samp>
+{{#suffix}}<span>{{suffix}}</span>{{/suffix}}
+{{#correct}}<span class="badge badge-success"><i class="fa fa-check" aria-hidden="true"></i> 100%</span>{{/correct}}
+{{#partial}}<span class="badge badge-warning"><i class="fa fa-circle-o" aria-hidden="true"></i> {{partial}}%</span>{{/partial}}
+{{#incorrect}}<span class="badge badge-danger"><i class="fa fa-times" aria-hidden="true"></i> 0%</span>{{/incorrect}}
+{{/parse_error}}
+{{#inline}}</span>{{/inline}}
+{{/submission}}
+
+{{#answer}}
+{{#label}}<span>{{label}}</span>{{/label}}
+<samp>{{a_tru}}</samp>
+{{#suffix}}<span>{{suffix}}</span>{{/suffix}}
+{{/answer}}
+
+{{#format}}
+<p>
+Your answer must be a string. Any symbolic expressions or numbers will be intepreted as a string.
+</p>
+{{/format}}
+
+{{#shortformat}}
+string
```

Comment:
It's probably cleaner just to hardcode this in Python? I know we have the pattern elsewhere, it just seems silly to have to get it out of the template.

Reply:
I can go one way or the other. I don't have an opinion about this, I was just following the same idea from other elements. Anyone else would like to comment on this? Should I make the change?

Reply:
I don't mind either way.